hr_timesheet_begin_end/models/account_analytic_line.py

```diff
@@ -15,16 +16,34 @@ class AccountAnalyticLine(models.Model):
     time_start = fields.Float(string="Begin Hour")
     time_stop = fields.Float(string="End Hour")
 
-    @api.constrains("time_start", "time_stop", "unit_amount")
-    def _check_time_start_stop(self):
-        for line in self:
-            value_to_html = self.env["ir.qweb.field.float_time"].value_to_html
-            start = timedelta(hours=line.time_start)
-            stop = timedelta(hours=line.time_stop)
-            if stop < start:
-                value_to_html(line.time_start, None)
-                value_to_html(line.time_stop, None)
+    # Override to be a computed field.
+    unit_amount = fields.Float(
+        compute="_compute_unit_amount",
+        store=True,
+        readonly=False,
+        # This default is a workaround for a bizarre situation: if a line is
+        # created with a time range but WITHOUT defining unit_amount, then you
+        # would expect unit_amount to be computed from the range. But this never
+        # happens, and it is instead set to default value 0. Subsequently the
+        # constraint _validate_unit_amount_equal_to_time_diff kicks in and
+        # raises an exception.
+        #
+        # By setting the default to None, the computation is correctly
+        # triggered. If nothing is computed, None falls back to 0.
```

Comment:
i think that the comment can be simpler. something like:

```suggestion
        # removing the default of 0.0 to ensure it is computed when not
        # provided. if not computed (because project_id is False), None is
        # automatically converted to 0.
```